### ISSUE TITLE
Record the TPM State of each node and if encryption enabled, backup the keys of each node into the report

### ISSUE THREAD

**Issue by CodeJACKz** [AsBuiltReport/AsBuiltReport.VMware.vSphere#101]
Since TPM should now be implemented for all deployments, can this detail be added to the report?

Link with example code:
https://vm.knutsson.it/2021/07/powercli-tpm-encryption-recovery-key-backup/

I got bitten by this after a recent deployment where the keys were not recorded and a node failed a couple of weeks after. Recovery would have been possible and faster had i recorded all the keys.

**Comment by tpcarman** [comment]
I'll look into adding the TPM information, however an as-built does not set or change a configuration, it simply documents and records the current configuration.

**Comment by CodeJACKz** [comment]
Thanks Tim. I only included them "set" commands to hint at a place to check if it was even enabled or not

**Comment by tpcarman** [comment]
Yeah thanks for that, I will likely use that to report on whether it is set or not, and probably add a health check for it too.
